### PR TITLE
Fix Arm(R) Corstone(TM)-300 FVP CI script

### PR DESCRIPTION
**Pull request by mansnils**
Benchmark target is not supported for armclang.

Also the cortex-m-* workflows are updated to correctly include PR in checkout, if tested from a PR.

BUG=https://github.com/tensorflow/tflite-micro/issues/2775#issuecomment-2481913933

### CODE REVIEW

File: tensorflow/lite/micro/tools/ci_build/test_cortex_m_corstone_300.sh [hunk 1]
```diff
@@ -43,13 +43,15 @@ readable_run make -f tensorflow/lite/micro/tools/make/Makefile clean
...
-readable_run make -j$(nproc) -f tensorflow/lite/micro/tools/make/Makefile \
-  CO_PROCESSOR=ethos_u \
-  OPTIMIZED_KERNEL_DIR=${OPTIMIZED_KERNEL_DIR} \
-  TARGET=${TARGET} \
-  TARGET_ARCH=${TARGET_ARCH} \
-  TOOLCHAIN=${TOOLCHAIN} \
-  GENERIC_BENCHMARK_MODEL_PATH=tensorflow/lite/micro/models/person_detect_vela.tflite \
-  GENERIC_BENCHMARK_ARENA_SIZE=`expr 150 \* 1024` \
-  run_tflm_benchmark
+# Run generic benchmark. Not supported for armclang - see comment in target makefile.
```
Comment (ddavis-2015) [inline tensorflow/lite/micro/tools/ci_build/test_cortex_m_corstone_300.sh:46]: @mansnils 

You mention a comment in the Generic Benchmark application makefile?  We would be very interested in hearing more about this issue.

Comment (ddavis-2015) [reply tensorflow/lite/micro/tools/ci_build/test_cortex_m_corstone_300.sh:46]: @mansnils 

Never mind, I found the makefile with the clang comment.  I now understand this change.

**Review by ddavis-2015 — APPROVED** [review]
@suleshahid 

Looks good to me.
Also, fixes the CI full-run build.

**Review by ddavis-2015 — COMMENTED** [review]
@mansnils @suleshahid 

While this fix looks correct, it still fails the ARM Compiler 6 Corstone 300 CI (ci:full_run).

**Comment by mansnils** [comment]
> @mansnils @suleshahid
> 
> While this fix looks correct, it still fails the ARM Compiler 6 Corstone 300 CI (ci:full_run).

That is strange. The fix should make this command pass: `tensorflow/lite/micro/tools/ci_build/test_cortex_m_corstone_300.sh armclang`
Although from the c:full_run log it looks like the original error message is still there.

**Comment by mansnils** [comment]
> > @mansnils @suleshahid
> > While this fix looks correct, it still fails the ARM Compiler 6 Corstone 300 CI (ci:full_run).
> 
> That is strange. The fix should make this command pass: `tensorflow/lite/micro/tools/ci_build/test_cortex_m_corstone_300.sh armclang` Although from the c:full_run log it looks like the original error message is still there.

@Evagan2018 Could it be that the Arm Compiler 6 workflow does not include the SHA (PR) during the test?

**Comment by MatthiasHertel80** [comment]
We are looking into the issue now.

**Comment by mansnils** [comment]
@suleshahid I suggest we merge it and if Arm Compiler 6 workflow still fails we can revert.

**Review by suleshahid — APPROVED** [review]

**Comment by suleshahid** [comment]
https://github.com/Mergifyio refresh

**Review by suleshahid — APPROVED** [review]

**Pull request merged**